.github/workflows/actionlint.yaml

```diff
@@ -0,0 +1,23 @@
+name: Actionlint
+
+on:
+  pull_request:
+  push:
+    branches: main
+  # allow "manual" triggering from automatic PRs
+  workflow_dispatch:
+
+jobs:
+  actionlint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      # Taken from https://github.com/rhysd/actionlint/blob/v1.7.7/docs/usage.md#use-actionlint-on-github-actions
+      - name: Download actionlint
+        id: get_actionlint
+        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
+        shell: bash
```

Comment:
This basically runs arbitrary code from a 3rd-party repo. When triggered with 'push' or 'workflow_dispatch', I'm pretty sure a GitHub Action runs with access to privileges (the 'permissive' set in https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

Could we lock down the token for this workflow? And disable `persist-credentials` for the checkout action above? (https://github.com/actions/checkout/)

Reply:
@raboof> set `persist-credentials` to false (thanks, didn't know about this one) and restricted `GITHUB_TOKEN`'s rights to the minimum :+1:

Reply:
> @raboof> set `persist-credentials` to false (thanks, didn't know about this one) and restricted `GITHUB_TOKEN`'s rights to the minimum 👍 See the last two commits.

You might be interested in https://github.com/woodruffw/zizmor that catches this. If you want an example of how it's already used in nixos/infra, https://github.com/NixOS/infra/blob/b464b1d87008ce8b61838a7d6ae55fa9b2cd5f11/.github/workflows/zizmor.yml

Reply:
Cool, didn't know about zizmor, thanks for sharing @Erethon :heart: